addons/binding/org.openhab.binding.geofence/src/main/java/org/openhab/binding/geofence/internal/BindingComponent.java

```diff
@@ -0,0 +1,264 @@
+        service = {ThingHandlerFactory.class, ManagedService.class}
+)
+public class BindingComponent extends BaseThingHandlerFactory implements ManagedService {
+    /**
+     * Class logger
+     */
+    private final Logger logger = LoggerFactory.getLogger(BindingComponent.class);
+
+    /**
+     * Discovery service registration
+     */
+    private ServiceRegistration<?> serviceRegistration = null;
+
+    /**
+     * Binding configuration
+     */
+    private BindingConfiguration config = new BindingConfiguration();
+
+    /**
+     * HTTP service reference
+     */
+    private HttpService httpService;
+
+    /**
+     * Endpoint called by tracker applications
+     */
+    private OwnTracksCallbackServlet otHTTPEndpoint;
+
+    /**
+     * Endpoint called by tracker applications
+     */
+    private GPSLoggerCallbackServlet glHTTPEndpoint;
+
+    /**
+     * Discovery service
+     */
+    private DeviceDiscoveryService discoveryService;
+
+    /**
+     * Reference
+     */
+    private ThingRegistry thingRegistry;
+
+    /**
+     * Notification broker
+     */
+    private NotificationBroker notificationBroker = new NotificationBroker();
+
+    /**
+     * Called by the framework to find out if thing type is supported by the handler factory.
+     *
+     * @param thingTypeUID Thing type UID
+     * @return True if supported.
+     */
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return BindingConstants.SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    /**
+     * Creates new handler for device.
+     *
+     * @param thing Device thing
+     * @return Handler instance
+     */
+    protected ThingHandler createHandler(Thing thing) {
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+        if (BindingConstants.THING_TYPE_DEVICE.equals(thingTypeUID)) {
+            String id = thing.getUID().getId();
+            discoveryService.removeTracker(id);
+            return new TrackerDeviceHandler(thing, this.config, notificationBroker);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Activate the binding. It starts the device discovery service and the HTTP callback endpoint.
+     *
+     * @param componentContext Component context.
+     */
+    protected void activate(ComponentContext componentContext) {
+        super.activate(componentContext);
+        logger.debug("Initializing the configuration.");
+        updateConfiguration(componentContext.getProperties());
+        updatePrimaryLocationFromSystemConfig();
+
+        logger.debug("Initializing discovery service");
+        this.discoveryService = new DeviceDiscoveryService();
+        this.serviceRegistration = this.bundleContext.registerService(DiscoveryService.class.getName(), this.discoveryService, new Hashtable<>());
+
+        logger.debug("Initializing callback servlets");
+        otHTTPEndpoint = new OwnTracksCallbackServlet(httpService, thingRegistry, discoveryService);
```

Comment:
Instead of passing the httpService to the servlet and then call activate, which does nothing more than registering the servlet on the HTTP service, you should move the registration to here and neither pass the service nor require the activate.